wire-runtime/src/commonMain/kotlin/com/squareup/wire/MessageSink.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire
+
+expect interface MessageSink<in T> {
+  fun send(message: T)
```

Comment:
I'm tempted to have this called `write` and the source one called `readOrNull` for some symmetry with Okio's sink/source. @swankjesse ?

Reply:
Mostly because send/receive seems to indicate network where this might be reading from memory or filesystem or somewhere else.

Reply:
I like `write()`.

It also makes it more clear that this operates on a value not a reference. Ie. if you write something and then mutate it, the sink doesn't observe the mutation.